### ISSUE TITLE
Unknown column 'state' in 'field list'

### ISSUE THREAD

**Issue by localheinz** [zendframework/modules.zendframework.com#251]
Following https://github.com/zendframework//pull/250 I tried to sign in into the app, which resulted in the following exception being displayed:

### Exception

`Zend\Db\Adapter\Exception\InvalidQueryException`

### File

`vendor/zendframework/zendframework/library/Zend/Db/Adapter/Driver/Pdo/Statement.php:244`

### Message

```
Statement could not be executed (42S22 - 1054 - Unknown column 'state' in 'field list')
```

### Stack Trace

```
#0 vendor/zf-commons/zfc-base/src/ZfcBase/Mapper/AbstractDbMapper.php(141): Zend\Db\Adapter\Driver\Pdo\Statement->execute()
#1 vendor/zf-commons/zfc-user/src/ZfcUser/Mapper/User.php(55): ZfcBase\Mapper\AbstractDbMapper->insert(Object(User\Entity\User), NULL, NULL)
#2 vendor/socalnick/scn-social-auth/src/ScnSocialAuth/Authentication/Adapter/HybridAuth.php(534): ZfcUser\Mapper\User->insert(Object(User\Entity\User))
#3 vendor/socalnick/scn-social-auth/src/ScnSocialAuth/Authentication/Adapter/HybridAuth.php(450): ScnSocialAuth\Authentication\Adapter\HybridAuth->insert(Object(User\Entity\User), 'github', Object(Hybrid_User_Profile))
#4 vendor/socalnick/scn-social-auth/src/ScnSocialAuth/Authentication/Adapter/HybridAuth.php(110): ScnSocialAuth\Authentication\Adapter\HybridAuth->githubToLocalUser(Object(Hybrid_User_Profile))
#5 [internal function]: ScnSocialAuth\Authentication\Adapter\HybridAuth->authenticate(Object(ZfcUser\Authentication\Adapter\AdapterChainEvent))
#6 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(468): call_user_func(Array, Object(ZfcUser\Authentication\Adapter\AdapterChainEvent))
#7 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(207): Zend\EventManager\EventManager->triggerListeners('authenticate', Object(ZfcUser\Authentication\Adapter\AdapterChainEvent), Object(Closure))
#8 vendor/zf-commons/zfc-user/src/ZfcUser/Authentication/Adapter/AdapterChain.php(56): Zend\EventManager\EventManager->trigger('authenticate', Object(ZfcUser\Authentication\Adapter\AdapterChainEvent), Object(Closure))
#9 vendor/zf-commons/zfc-user/src/ZfcUser/Controller/UserController.php(139): ZfcUser\Authentication\Adapter\AdapterChain->prepareForAuthentication(Object(ZF\ContentNegotiation\Request))
#10 vendor/zendframework/zendframework/library/Zend/Mvc/Controller/AbstractActionController.php(83): ZfcUser\Controller\UserController->authenticateAction()
#11 [internal function]: Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#12 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(468): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#13 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(207): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#14 vendor/zendframework/zendframework/library/Zend/Mvc/Controller/AbstractController.php(117): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#15 vendor/zendframework/zendframework/library/Zend/Mvc/Controller/Plugin/Forward.php(142): Zend\Mvc\Controller\AbstractController->dispatch(Object(ZF\ContentNegotiation\Request), Object(Zend\Http\PhpEnvironment\Response))
#16 vendor/socalnick/scn-social-auth/src/ScnSocialAuth/Controller/UserController.php(144): Zend\Mvc\Controller\Plugin\Forward->dispatch('zfcuser', Array)
#17 vendor/zendframework/zendframework/library/Zend/Mvc/Controller/AbstractActionController.php(83): ScnSocialAuth\Controller\UserController->providerAuthenticateAction()
#18 [internal function]: Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#19 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(468): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#20 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(207): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#21 vendor/zendframework/zendframework/library/Zend/Mvc/Controller/AbstractController.php(117): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#22 vendor/zendframework/zendframework/library/Zend/Mvc/DispatchListener.php(114): Zend\Mvc\Controller\AbstractController->dispatch(Object(ZF\ContentNegotiation\Request), Object(Zend\Http\PhpEnvironment\Response))
#23 [internal function]: Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#24 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(468): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#25 vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(207): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#26 vendor/zendframework/zendframework/library/Zend/Mvc/Application.php(313): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#27 //public/index.php(21): Zend\Mvc\Application->run()
#28 {main}
```

Given that I'm working with a database dump generated by @GeeH yesterday, I understand this means that currently nobody can sign in who hasn't signed in before. 

I added a column `state` locally (just used `VARCHAR(255)`), then signing in worked.

Checking with the database again:

```
{
	"data":
	[
		{
			"user_id": 99999,
			"username": null,
			"email": "email@example.com",
			"display_name":"Andreas Möller",
			"password": "githubToLocalUser",
			"photo_url": null,
			"created_at": "2014-12-06 19:16:03",
			"state": null
		}
	]
}
```

So apparently:

* `username` was not fetched properly
* `photo_url` wasn't fetched
* `email` was fetched, but not my primary email address, unfortunately
* `state` wasn't updated to whatever it should be, unless `null` is what it should be

**Comment by gerardZf2Comments** [comment]
The comiit 7870422 changes the module_id column of module table from being auto incrementing. Why was it done? Does it cause an error when adding a new module?

**Comment by gianarb** [comment]
I don't remember but maybe this bug is a fix ( :boom: ) of this #206..
@ins0 do you remember? :D

**Comment by ins0** [comment]
don't know but some of the changes improve the sql but some changes like the ``auto_increment`` looks like a misstake to me - can't be a fix for #206 as it is a 4 months old commit. @Danielss89 may have some ìnfo about these changes back in the days?

**Comment by localheinz** [comment]
Closing as I just added the column to the table a few minutes ago.
